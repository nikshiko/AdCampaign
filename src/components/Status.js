import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { Icon } from 'antd'

import { isDateInRange } from '../utils'
import { active, inactive, dateFormat } from '../constants'

export default function StatusComponent({startDate, endDate}) {
	let status = inactive
    
	if(isDateInRange(moment().format(dateFormat),startDate,endDate)) {
		status= active
	}

	return (
		<React.Fragment>
			<Icon
				type={status !== active ? 'minus-circle': 'check-circle'}
				style={{color: status !== active? '#ff0000' : '#008000'}} />
			{`\u00A0${status}`}
		</React.Fragment>
	)  
}

StatusComponent.propTypes = {
	/**
     * The start date used to determine the status of the campaign
     */
	startDate: PropTypes.string.isRequired,
	/**
     * The end date used to determine the status of the campaign
     */
	endDate: PropTypes.string.isRequired
}

