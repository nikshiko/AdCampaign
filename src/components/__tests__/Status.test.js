import React from 'react'
import { render } from '@testing-library/react'
import { toMatchDiffSnapshot } from 'snapshot-diff'

expect.extend({ toMatchDiffSnapshot })
import Status from '../Status'

describe('Status component',() => {
	it('should render the active and inactive status without crashing',() => {
		const { asFragment } = render(<Status startDate="06/20/2018" endDate="07/24/2019" />)
		const firstRender = asFragment()
		const { asFragment: newFragment } = render(<Status startDate="06/20/2018" endDate="06/21/2019" />)
		expect(newFragment()).toMatchDiffSnapshot(firstRender)
	})
})