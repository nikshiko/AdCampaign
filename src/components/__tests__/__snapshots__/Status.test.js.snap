// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status component should render the active and inactive status without crashing 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <i
-     aria-label=\\"icon: minus-circle\\"
-     class=\\"anticon anticon-minus-circle\\"
-     style=\\"color: rgb(255, 0, 0);\\"
+     aria-label=\\"icon: check-circle\\"
+     class=\\"anticon anticon-check-circle\\"
+     style=\\"color: rgb(0, 128, 0);\\"
    >
      <svg
        aria-hidden=\\"true\\"
        class=\\"\\"
-       data-icon=\\"minus-circle\\"
+       data-icon=\\"check-circle\\"
        fill=\\"currentColor\\"
        focusable=\\"false\\"
        height=\\"1em\\"
        viewBox=\\"64 64 896 896\\"
        width=\\"1em\\"
      >
        <path
-         d=\\"M696 480H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h368c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8z\\"
+         d=\\"M699 353h-46.9c-10.2 0-19.9 4.9-25.9 13.3L469 584.3l-71.2-98.8c-6-8.3-15.6-13.3-25.9-13.3H325c-6.5 0-10.3 7.4-6.5 12.7l124.6 172.8a31.8 31.8 0 0 0 51.7 0l210.6-292c3.9-5.3.1-12.7-6.4-12.7z\\"
        />
        <path
          d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z\\"
        />
      </svg>
    </i>
-    Inactive
+    Active
  </DocumentFragment>"
`;
